src/test/scala/re2s/ExecTest.scala

```diff
@@ -0,0 +1,686 @@
+              case ':' =>
+                val i = flag.indexOf(':', 1)
+                if (i < 0) {
+                  System.err.format("skip: %s\n", line)
+                  continue = true //todo: continue is not supported
+                }
+                if (!continue) flag = flag.substring(1 + i + 1)
+
+              case 'C' | 'N' | 'T' | '0' | '1' | '2' | '3' | '4' | '5' | '6' |
+                  '7' | '8' | '9' =>
+                System.err.println("skip: %s\n".format(line))
+                continue = true
+              case _ =>
+            }
+            if (!continue) {
+              // Can check field count now that we've handled the myriad comment
+              // formats.
+              if (field.size < 4) {
+                System.err.println(
+                  "%s:%d: too few fields: %s\n".format(file, lineno, line))
+                nerr += 1
+                continue = true
+              }
+              if (!continue) {
+                // Expand C escapes (a.k.a. Go escapes).
+                if (flag.indexOf('$') >= 0) {
+                  var f = "\"" + field.get(1) + "\""
+                  try field.set(1, Strconv.unquote(f))
+                  catch {
+                    case e: Exception =>
+                      System.err.println(
+                        "%s:%d: cannot unquote %s\n".format(file, lineno, f))
+                      nerr += 1
+                  }
+                  f = "\"" + field.get(2) + "\""
+                  try field.set(2, Strconv.unquote(f))
+                  catch {
+                    case e: Exception =>
+                      System.err.println(
+                        "%s:%d: cannot unquote %s\n".format(file, lineno, f))
+                      nerr += 1
+                  }
+                }
+                //   Field 2: the regular expression pattern; SAME uses the pattern from
+                //     the previous specification.
+                if (field.get(1) == "SAME") field.set(1, lastRegexp)
+                lastRegexp = field.get(1)
+                //   Field 3: the string to match.
+                val text = field.get(2)
+                //   Field 4: the test outcome...
+                val shouldCompileMatch = Array(false, false)
+                // in/out param to parser
+                var pos: util.List[Integer] = null
+                try pos = parseFowlerResult(field.get(3), shouldCompileMatch)
+                catch {
+                  case e: Exception =>
+                    System.err.println(
+                      "%s:%d: cannot parse result %s\n"
+                        .format(file, lineno, field.get(3)))
+                    nerr += 1
+                    continue = true
+                }
+                //   Field 5: optional comment appended to the report.
+                // Run test once for each specified capital letter mode that we support.
+                if (!continue) {
```

Comment:
for complex break code: https://www.scala-lang.org/api/current/scala/util/control/Breaks.html